.travis.yml

```diff
@@ -0,0 +1,40 @@
+language: go
+
+go:
+  - "1.15.5"
+
+os:
+  - linux
+
+dist: xenial
+
+cache:
+  directories:
+    - $HOME/.cache/go-build
+    - $HOME/gopath/pkg/mod
+
+env:
+  global:
+    - GO111MODULE=on
+    - GORACE="halt_on_error=1"
+
+before_script:
+  - set -eo pipefail
+  - go get -u golang.org/x/tools/cmd/goimports
+script:
+  # We could merge the two commands behind for efficiency, but it would give less comprehensive errors
+  # Check files syntax and basic formatting
+  - echo "Files with syntax or formatting errors :"
+  - find . -name "*.go" -execdir gofmt -l -s '{}' \+
+  # Check files formatting, including imports
+  - echo "Files with formatting errors :"
+  - find . -name "*.go" -execdir goimports -l '{}' \+
+  # Run the test suite
```

Comment:
At this point we should check if the files were modified by the linter and/or formatter with something like

`[[ $(git ls-files --modified | wc --lines) == 0 ]] && true || ( echo -e '\nFix lint/format locally and then commit.\n' && exit 1 )`

or similar.

Reply:
The linter with `-l` option does not modify the files but only reports the non complying files, so I went for another solution